pypose/function/geometry.py

```diff
@@ -159,3 +175,210 @@ def reprojerr(points, pixels, intrinsics, extrinsics=None):
+        necessarily aligned with the index of the input point cloud.
+
+    Returns:
+        ``torch.return_types.topk (values: torch.Tensor, indices: torch.Tensor)``: The
+        named tuple of (values, indices).
+
+        ``values``: The ord-norm distance between each point in pc1 and its sorted k
+        nearest neighbors in pc2. The shape is (..., N1, k).
+
+        ``indices``: The index of the k nearest neighbor points in pc2.
+        The shape is (..., N1, k).
+
+    Example:
+        >>> import torch, pypose as pp
+        >>> pc1 = torch.tensor([[9., 2., 2.],
+        ...                     [1., 0., 2.],
+        ...                     [0., 1., 1.],
+        ...                     [5., 0., 1.],
+        ...                     [1., 0., 1.],
+        ...                     [5., 5., 3.]])
+        >>> pc2 = torch.tensor([[1., 0., 1.],
+        ...                     [1., 6., 2.],
+        ...                     [5., 1., 0.],
+        ...                     [9., 0., 2.]])
+        >>> pp.knn(pc1, pc2)
+        torch.return_types.topk(
+        values=tensor([[2.0000],
+                [1.0000],
+                [1.4142],
+                [1.4142],
+                [0.0000],
+                [4.2426]]),
+        indices=tensor([[3],
+                [0],
+                [0],
+                [2],
+                [0],
+                [1]]))
+        >>> pp.knn(pc1, pc2, k=2, ord=2)
+        torch.return_types.topk(
+        values=tensor([[2.0000, 4.5826],
+                [1.0000, 4.5826],
+                [1.4142, 5.0990],
+                [1.4142, 4.0000],
+                [0.0000, 4.2426],
+                [4.2426, 5.0000]]),
+        indices=tensor([[3, 2],
+                [0, 2],
+                [0, 2],
+                [2, 0],
+                [0, 2],
+                [1, 2]]))
+        >>> pp.knn(pc1, pc2, k=2, ord=2).values
+        tensor([[2.0000, 4.5826],
+                [1.0000, 4.5826],
+                [1.4142, 5.0990],
+                [1.4142, 4.0000],
+                [0.0000, 4.2426],
+                [4.2426, 5.0000]])
+    '''
+    diff = pc1.unsqueeze(-2) - pc2.unsqueeze(-3)
+    dist = torch.linalg.norm(diff, dim=dim, ord=ord)
+    neighbors = dist.topk(k, dim=dim, largest=largest, sorted=sorted)
+    return neighbors
+
+
+def svdtf(pc1, pc2):
+    r'''
+    Computes the rigid transformation ( :math:`SE(3)` ) between two sets of associated
+    points using Singular Value Decomposition (SVD).
+
+    Args:
+        pc1 (``torch.Tensor``): The coordinates of the first set of points.
+            The shape has to be (..., N, 3).
+        pc2 (``torch.Tensor``): The coordinates of the second set of points.
+            The shape has to be (..., N, 3).
+
+    Returns:
+        ``LieTensor``: The rigid transformation matrix in ``SE3Type``  that
+        minimizes the mean squared error between the input point sets.
+
+    Warning:
+        The number of points N has to be the same for both point clouds.
```

Comment:
suggest: 'The input point clouds have to be associated and have the same number of points N.'